### PR TITLE
Update lsf script

### LINKED ISSUE #56: Replace `absolute-path` to `relative-path`
## What analysis module should be updated and why?

Update `lsf` files per module to run the container and replace `absolute-path` to `relative-path`.

### LINKED ISSUE #57: To update `fastqc-analysis` and `upstream-analysis` to run within the containeer
## What analysis module should be updated and why?

To update `fastqc-analysis` and `upstream-analysis` to run within the containeer.

### LINKED ISSUE #69: Streamline instructions in the main repo
## What analysis module should be updated and why?

Streamline instructions in the main repo.

### PR DESCRIPTION
**Pull request by AntoniaChroni**
# Purpose/implementation Section

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## What scientific question/module of analysis is your analysis addressing?

To update `fastqc-analysis` and `upstream-analysis` to run within the containeer.

## What GitHub issue does your pull request address?

Closes #57.
Closes #69.

## Directions for reviewers 
Tell potential reviewers what kind of feedback you are soliciting.

### Which areas should receive a particularly close look?

Please submit the `lsf` scripts as described in the `README` files and confirm the successful run for the following modules:

- [x] `fastqc-analysis`
- [x] `upstream-analysis`

Please also check the `README` file in the main directory for clarity:

- [x] `README` file

## Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.

## Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by AntoniaChroni** [comment]
@sharonfreshour reported an issue while running the `upstream-analysis` module in the container. For reference, the module runs smoothly outside of the container. The error comes up after removing these modules: `module load R/4.4.0`, `module load pandoc/3.2`, and `module load texlive/20240326`.  I am uncertain about the issue at this moment.

I tried the following so far:

- I updated the `run-container/run-test-other-dependencies.sh` (see 09e991584df269f76562fa265b3fcae1613c7059),  and the following came up:

`Python 3.10.12
cellranger cellranger-8.0.1
FastQC v0.11.9
multiqc, version 1.25.1
pandoc 3.2
Features: +server +lua
Scripting engine: Lua 5.4
User data directory: /home/achroni/.local/share/pandoc
Copyright (C) 2006-2024 John MacFarlane. Web: https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
FATAL:   "texlive": executable file not found in $PATH`

- Moreover, I created a bash file to run the Rmd (see 77893c3b4c31cc4d0d0928093a973da2d3a58018) and failed to create pdf file.

@walidabualafia Any insights on this?

**Comment by AntoniaChroni** [comment]
@sharonfreshour, I removed some typos and streamlined the `README` file in the main repo for this branch. See 90a47867be669963e983d26ff5d46427521c665d. Please review this file for clarity as well.

**Comment by AntoniaChroni** [comment]
> @sharonfreshour reported an issue while running the `upstream-analysis` module in the container.
> 
> ```
> processing file: 01A_run_seurat_qc.Rmd
> Error in `x()`:
> ! plot.new has not been called yet
> Backtrace:
>   1. rmarkdown::render(...)
>   2. rmarkdown::render(...)
>   3. knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
>   4. knitr:::process_file(text, output)
>   7. knitr:::process_group(group)
>      ...
>  15. base::withRestarts(...)
>  16. base (local) withRestartList(expr, restarts)
>  17. base (local) withOneRestart(withRestartList(expr, restarts[-nr]), restarts[[nr]])
>  18. base (local) docall(restart$handler, restartArgs)
>  20. evaluate (local) fun(base::quote(`<smplErrr>`))
> 
> Quitting from lines 267-289 [plot-data-before-filtering] (01A_run_seurat_qc.Rmd)
> Execution halted
> ```
> 
> For reference, the module runs smoothly outside of the container. The error comes up after removing these modules: `module load R/4.4.0`, `module load pandoc/3.2`, and `module load texlive/20240326`. I am uncertain about the issue at this moment.
> 
> I tried the following so far:
> 
> * I updated the `run-container/run-test-other-dependencies.sh` (see [09e9915](https://github.com/stjude-dnb-binfcore/sc-rna-seq-snap/commit/09e991584df269f76562fa265b3fcae1613c7059)),  and the following came up:
> 
> ```
> Python 3.10.12
> cellranger cellranger-8.0.1
> FastQC v0.11.9
> multiqc, version 1.25.1
> pandoc 3.2
> Features: +server +lua
> Scripting engine: Lua 5.4
> User data directory: ./.local/share/pandoc
> Copyright (C) 2006-2024 John MacFarlane. Web: https://pandoc.org
> This is free software; see the source for copying conditions. There is no
> warranty, not even for merchantability or fitness for a particular purpose.
> FATAL:   "texlive": executable file not found in $PATH
> ```
> 
> * Moreover, I created a bash file to run the Rmd (see [77893c3](https://github.com/stjude-dnb-binfcore/sc-rna-seq-snap/commit/77893c3b4c31cc4d0d0928093a973da2d3a58018)) and failed to create pdf file.
> 
> @walidabualafia Any insights on this?

Two separate issues resulted in these errors.

1. Improper/incomplete installation of `texlive` suite. Docker image has been updated, see f7c7d0bc39951918ee286c1b63d3df939393cb19.
2. A bug from a variable in the functions in the `function-calculate-qc-metrics.R` script. This has been addressed here 201ae1359eab8437b6089053b84e490093f928c7. 

@sharonfreshour Please run the lsf scripts for both `fastqc-analysis` and `upstream-analysis` modules.

**Review by walidabualafia — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -54,31 +43,30 @@ cd /home/rstudio/sc-rna-seq-snap/analyses/<module_of_interest>
...
+Then navigate to your `./sc-rna-seq-snap` project repo and ensure you are at the `main` branch. If not, you will need to `git checkout` to the main branch.
 ```
 git branch
 git checkout main
 ```
 
 Finally, `git pull` to get the most updated changes and code in your project repo. 
-
 ```
 git pull
```
Comment (sharonfreshour) [inline README.md:67]: If I sync and do a git pull for the forked repo after updating the yaml file to my paths, I get an error saying I either need to commit or stash (or restore) the changes to my yaml before completing the pull. So maybe either the info for syncing and pulling the forked repo should be moved back up above the section to run the code in the repo. Or else maybe add instructions for how to handle the yaml file.

Comment (AntoniaChroni) [reply README.md:67]: This has been addressed here 314de0925e8d728e2e313a1c80ee90118bd519bf.

**Review by sharonfreshour — APPROVED** [review]
Both fastqc and upstream analysis completed successfully for me!

**Pull request merged**